hw/opentitan/ot_hmac.c

```diff
@@ -326,16 +557,24 @@ static void ot_hmac_compute_digest(OtHMACState *s)
     if (s->regs->cfg & R_CFG_HMAC_EN_MASK) {
         ot_hmac_sha_done(s);
 
-        uint64_t opad[8u];
+        size_t key_length_b = ot_hmac_get_key_bytes(s);
+        size_t block_size_b = ot_hmac_get_block_size_bytes(s);
+        /* If the key is smaller than the block size, we must pad it to the
+        right with 0s. */
+        size_t pad_length_b = MAX(key_length_b, block_size_b);
+        size_t pad_length_w = pad_length_b / sizeof(uint32_t);
+        uint32_t opad[OT_HMAC_MAX_KEY_LENGTH / sizeof(uint32_t)];
         memset(opad, 0, sizeof(opad));
-        memcpy(opad, s->regs->key, sizeof(s->regs->key));
-        for (unsigned i = 0; i < ARRAY_SIZE(opad); i++) {
-            opad[i] ^= 0x5c5c5c5c5c5c5c5cull;
+        memcpy(opad, s->regs->key, key_length_b);
+        for (unsigned i = 0; i < pad_length_w; i++) {
```

Comment:
I was mostly aiming to leave the existing code as is where possible, but I'll change the name of this variable to match current styling. Based on the context of the actual values I think the type widths should be fine here but as good practice I'll change the `unsigned` to a `size_t`.

Reply:
Please do. HMAC is one of the oldest IP we've implemented, and we learnt and refined the rules since then, and we have not updated the code. As you're actively working on it, this is a nice opportunity to align the code. Sorry for adding some workload here.